templates/workloads/common/cf/autoscaling.yml

```
@@ -54,4 +54,27 @@ AutoScalingPolicyECSServiceAverageMemoryUtilization:
       ScaleOutCooldown: 60
       TargetValue: {{.Autoscaling.Memory}}
 {{- end}}
-{{- end}}
\ No newline at end of file
+{{- end}}
+
+DynamicDesiredCountAction:
+  Type: Custom::DynamicDesiredCountFunction
+  Properties:
+    ServiceToken: !GetAtt DynamicDesiredCountFunction.Arn
+    Cluster:
+      Fn::ImportValue:
+        !Sub '${AppName}-${EnvName}-ClusterId'
+    Family: !Join ['', [!Ref AppName, '-', !Ref EnvName, '-', !Ref WorkloadName]]
+    DesiredCount: !Ref TaskCount
+    UpdateID: {{.DesiredCountLambdaUpdateID}}
+
+DynamicDesiredCountFunction:
+  Type: AWS::Lambda::Function
+  Properties:
+    Code:
+      ZipFile: |
+        {{.DesiredCountLambda}}
+    Handler: "index.desiredCountHandler"
+    Timeout: 600
+    MemorySize: 512
+    Role: !GetAtt 'CustomResourceRole.Arn'
+    Runtime: nodejs10.x
```

We can bump this now
```suggestion
    Runtime: nodejs12.x
```